File: fsharp/guessing-game/GuessingGame.fs
module GuessingGame

let private target = 42;

let reply (guess: int): string =
    match guess - target with
    | 0 -> "Correct"
    | n when abs n = 1 -> "So close"
    | n when n <= -2 -> "Too low"
    | _ -> "Too high"

File: haskell/guessing-game/src/GuessingGame.hs
module GuessingGame (reply) where

reply :: Int -> String
reply guess
  | guess == answer = "Correct"
  | guess - 1 == answer || guess + 1 == answer = "So close"
  | guess < answer = "Too low"
  | otherwise = "Too high"
  where answer = 42

